src/devices/bus/nes/bootleg.cpp

```diff
@@ -284,15 +284,7 @@ void nes_smb3p_device::device_start()
 
 void nes_smb3p_device::pcb_reset()
 {
-	m_chr_source = m_vrom_chunks ? CHRROM : CHRRAM;
-	prg8_89((m_prg_chunks << 1) - 1);
-	prg8_ab(0);
-	prg8_cd(0);
-	prg8_ef((m_prg_chunks << 1) - 1);
-	chr8(0, m_chr_source);
-
-	m_irq_enable = 0;
-	m_irq_count = 0;
+	// registers not cleared or initialized at reset
```

Comment:
If you’re not clearing on reset, do you need to move stuff to start to ensure it begins in a deterministic state?

Reply:
Hmm, I looked into this a bit more. The short of it is that every* 8K code bank has the same reset vector code which immediately points 0x8000 to somewhere, jumps there, and sets things to a sane state. The PCB is odd for a NES cart and is loaded with discrete circuits.

* Except for one page. Game will fail if bank 24 is loaded into 0xE000 at power up/reset.

So there are at least two options:
(1) Don't be explicit, rely on how nes_slot.cpp initializes things (PRG and CHR banks set to start of ROMs I believe), and it will just work based on the common reset vectors. This option is what the current proposed change with the above empty pcb_reset does.
(2) Be explicit. Override nes slot's pcb_start function, not seen here, and initialize once there. device_start can't be used because prg8_xx helpers access stuff that gets initialized later by pcb_start (i.e. we'd segfault in device_start). NesDev reverse engineering does say that registers on the board reliably power on with all 0xFF. This option documents the best-to-date knowledge of the PCB (but effectively does the same as (1) after a few machine cycles).

Reply:
I went all executive and explicitly initialized things with choice (2). It only adds a few lines of mostly boilerplate.